10.1.0/startup/generateStartupCommand.js

```diff
@@ -1,78 +1,180 @@
-
-const DEFAULTAPP = "/opt/startup/default-static-site.js";
 const CMDFILE = "/opt/startup/startupCommand";
 
-var httpLoggingEnabled = process.env.HTTP_LOGGING_ENABLED;
-httpLoggingEnabled = (typeof httpLoggingEnabled !== 'undefined'
-    && httpLoggingEnabled !== null
-    && (httpLoggingEnabled.toLowerCase() === 'true' || httpLoggingEnabled.toLowerCase() === '1'))
-
-var roleInstanceId = '';
-if (typeof process.env.WEBSITE_ROLE_INSTANCE_ID !== 'undefined'
-    && process.env.WEBSITE_ROLE_INSTANCE_ID !== null) {
-    roleInstanceId = process.env.WEBSITE_ROLE_INSTANCE_ID;
+function getJsStartupCommand(jsFile) {
+    if (process.env.APPSVC_REMOTE_DEBUGGING == "TRUE") {
+        if (process.env.APPSVC_REMOTE_DEBUGGING_BREAK == "TRUE") {
+            return "node --inspect-brk=0.0.0.0:" + process.env.APPSVC_TUNNEL_PORT + " " + jsFile;
+        } else {
+            return "node --inspect=0.0.0.0:" + process.env.APPSVC_TUNNEL_PORT + " " + jsFile;
+        }
+    } else {
+        // Run with pm2
+        return "pm2 start " + jsFile + " --no-daemon";
+    }
 }
 
-var startupCommand = fs.readFileSync(CMDFILE, 'utf8').trim();
+function getStartupCommand(CMDFILE) {
+    const CUSTOM_STARTUP_CMD_FLAG = "/opt/startup/CUSTOM_STARTUP_CMD_FLAG";
+    fs.writeFileSync(CUSTOM_STARTUP_CMD_FLAG, "FALSE");
+    const PACKAGE_JSON_FLAG = "/opt/startup/PACKAGE_JSON_FLAG";
+    fs.writeFileSync(PACKAGE_JSON_FLAG, "FALSE");
+    const PROCESS_JSON_FLAG = "/opt/startup/PROCESS_JSON_FLAG";
+    fs.writeFileSync(PROCESS_JSON_FLAG, "FALSE");
 
-const CUSTOM_STARTUP_CMD_FLAG = "/opt/startup/CUSTOM_STARTUP_CMD_FLAG";
-fs.writeFileSync(CUSTOM_STARTUP_CMD_FLAG, "FALSE");
-if (startupCommand) {
-    fs.writeFileSync(CUSTOM_STARTUP_CMD_FLAG, "TRUE"); // set CUSTOM_STARTUP_CMD_FLAG for remote debugging
-}
+    const DEFAULTAPP = "/opt/startup/default-static-site.js";
+    var userStartupCommand = fs.readFileSync(CMDFILE, 'utf8').trim();
 
-// No user-provided startup command, check for scripts.start
-const PACKAGE_JSON_FLAG = "/opt/startup/PACKAGE_JSON_FLAG";
-fs.writeFileSync(PACKAGE_JSON_FLAG, "FALSE");
-if (!startupCommand) {
-    var packageJsonPath = "/home/site/wwwroot/package.json";
-    var json = fs.existsSync(packageJsonPath) && JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'))
-    if (typeof json == 'object' && typeof json.scripts == 'object' && typeof json.scripts.start == 'string') {
-        console.log("Found scripts.start in package.json")
-        startupCommand = 'npm start';
-        fs.writeFileSync(PACKAGE_JSON_FLAG, "TRUE"); // set PACKAGE_JSON_FLAG for remote debugging
+    if (userStartupCommand) {
+        fs.writeFileSync(CUSTOM_STARTUP_CMD_FLAG, "TRUE"); // set CUSTOM_STARTUP_CMD_FLAG for remote debugging
+    }
+    
+    // contains spaces, so multipart command
+    // eg: npm start
+    if (userStartupCommand.indexOf(" ") != -1) {
+        console.log("Found command: "+ userStartupCommand);
+        return userStartupCommand;
+    }
```

Comment:
if AppCommandLine has a space then its a command and execute as is

Reply:
How wold a command like so without spaces `./bin/www` or `./bin/start.sh` be processed?

Reply:
it would be handle at L141